docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,102 @@
+
+A delivery channel will have a generic api that can be consumed by the Management Api and be able to transform data provided in the Notification into a format that is specific to that communication type (this will be discussed in a sequential adr). The result of this call will be a callback uri/null depending on how the delivery channel has been implemented.
+
+### Delivery Channel configuration 
+
+A delivery channel will also need the configuration of the underlying platform to be stored somewhere (api keys, secrets, etc). This will be stored in the `Marain.UserNotifications` keyvault. If a delivery channel needs to be used, there are two things that need to be added/modified:
+1. Modify the `Marain.UserNotifications` key vault secrets setion with the correct key/value pair (this is described in more details below).
+2. Update the configuration for the delivery channel in the management api to specify which communication types will use the delivery channel. 
+
+Keys for the different platforms in the delivery channels will be stored in a format as follows:
+`<TenantId>-<key_name>`
+ an example with this format would look like:
+`75b9261673c2714681f14c97bc0439fb00000000000000000000000000000000-ApiMasterSecret`
+
+A key vault will be associated to each channel in the `Marain.UserNotifications` These key vaults will be created with the following format:
+`deliverychannel-<ChannelName>`
+an example with this format would be:
+`deliverychannel-airship`
+
+The Delivery Channel will also have to have it's own kind of configuration specifying the supported Communication Types. These will be hardcoded values and created when the coding of the new delivery channel is done. 
+
+### Delivery Channel usage
+
+The delivery channel will need to transform an object into the equivalent api model used to send a message by the chosen platform (twilio, airship, send grid etc). Some notifications will have different headings, bodys, generic templates that will have to be transformed into a format that is suitable and also might have different ways of targetting certain users depending on how the users are registered with the platforms. (eg. twilio this will be a number, for web push this could be a mixture of business id and email, etc). 
+
+A solution to this would be to create a generic type object for each commucication type (eg. smsobject, emailobject, webpushobject) that is provided by the management api to the delivery channel, and then the delivery channel can map a generic communication type object into the required api model. This will prove to be simpler when integrating new delivery channels in the future as most of the information will already be there.
```

Comment:
Is this the approach that's being taken? If so, reword to reflect that, and possibly give an example to illustrate how that would work.

Reply:
And are we are talking about "schema" for "payloads" for different communication types, rather than "generic type objects" because "generic", "type", and "object" all having pre-existing meanings!

Reply:
It is the approach that has been taken. 

I will add an example and remove the ambiguity.

Reply:
@jongeorge1 The fields within the objects will be templating, so it's sort of templating.